### DISCUSSION TITLE
First Impressions, possible nitpicks ahead!

### DISCUSSION THREAD

**Discussion by GasimoCodes** [ANU-CHEEKI-BREEKI/com.anupackages.debugconsole#7] [General]
I am using the latest asset store version, running Unity 2023.2 and these are my impressions so far. Keep in mind I have not used the packages for long so it truly is just first impressions:

- Looking at the docs, the core appears to be very powerful. Thats a big + considering its a MIT 1-person repo. Awesome work!
- The previous asset store version managed to break half of my project several months ago, I am happy to say that the new one has none of those issues and imports fine.
- The AssetStore import might be misconfigured, as the package did not get imported into packages and ended up in Assets/ANU/... This does not hinder the usability, but it appears to not be intentional either.
- The front-end UI of the console seems to be having some issues with layouting. This is extremely noticeable while scrolling or issuing new commands. For instance:
  - The bottommost text in the console is often not visible while scrolling up and then back down using the scrollbar
  - Text tends to jarringly pop in and out around as the layout gets redrawn during scrolling
  - Issuing command or scrolling 'out of bounds' (using drag) and letting go causes the console text to jitter up and down for about half a second.
 - While writing some commands (not executing them), its entirely possible to be getting ArgumentNullExceptions with every letter I write, flooding the command line. For instance test-commands.set-string-from-list-default <write anything, even spaces here> throws exceptions with every char written. 
 
Just to reiterate, its an awesome work you did and those are just some first impression nit-picks.
 
PS: Is it possible to plug in a runtime generated array as possible autocomplete values for an argument of a command? 
For instance a `teleport <entity> <location>` could use a location/entity name/id which would be autocompleted/suggested? I gave the docs a glance and I did see some compile-time autocomplete flags but I dont recall any runtime examples.

**Comment by ANU-CHEEKI-BREEKI** [comment]
Hi [GasimoCodes](https://github.com/GasimoCodes), and thank you for your feedback! It's really helpful in improving the asset. I personally use it in my own projects, but even then, I don't always catch all the issues.

`"Is it possible to plug in a runtime generated array?"`
Yes, please check the documentation for the [[OptValDynamic]](https://github.com/ANU-CHEEKI-BREEKI/com.anupackages.debugconsole?tab=readme-ov-file#attributes) attribute (there's a spoiler with an example there).

In general, you need to create a command that returns a collection of strings. This can be any implementation of IEnumerable<string> (array, list, etc.).

Then, add an argument to the method parameter where you want to display this list as suggestions, specifying the exact name of the command.

The asset will call the previously created command to get the list of strings and show them to you.

Thanks again for the feedback!
I would be very grateful if you could rate the asset in the asset store and leave a review there. 
Cheers and happy coding! 🚀
